p2p/peer.go

```diff
@@ -87,6 +87,8 @@ func newPeer(pc peerConn, nodeInfo NodeInfo,
 type PeerConfig struct {
 	AuthEnc bool `mapstructure:"auth_enc"` // authenticated encryption
 
+	Dial func(addr *NetAddress, config *PeerConfig) (net.Conn, error)
```

Comment:
I'd rather not have functions inside config.

Reply:
maybe we can Have `badAddr`, which returns error on `DialTimeout`. This will require to inroduce NetAddress interface.

Reply:
I think better would be to set a flag on the remotePeer to allow it to reject the the connection in its accept routine. We should try to avoid putting methods in config structs like this

Reply:
Maybe we can have a the `Dial` func somewhere else? It seems more convenient to have a configurable dial behavior for testing, for e.g it could be used for slow connections / timeouts.

Reply:
Maybe `Switch`?